csharp/lib/src/lib.rs

```diff
@@ -26,37 +29,75 @@ pub struct Client {
     runtime: Runtime,
 }
 
-fn create_connection_request(
-    host: String,
-    port: u32,
-    use_tls: bool,
+pub unsafe fn node_addresses_to_proto(
+    data: *const *const NodeAddress,
+    len: usize,
+) -> Vec<connection_request::NodeAddress> {
+    Vec::from_raw_parts(data as *mut NodeAddress, len, len).iter().map(|addr| {
+        let mut address_info = connection_request::NodeAddress::new();
+        address_info.host = CStr::from_ptr(addr.host).to_str().unwrap().into();
+        address_info.port = addr.port;
+        address_info
+    }).collect()
+/*
+    from_raw_parts(data, len)
+        .iter()
+        .map(|ptr| Box::leak(Box::from_raw(*ptr as *mut NodeAddress)))
+        .collect()
+ */
+}
+
+unsafe fn create_connection_request(
+    config: *const ConnectionRequest,
 ) -> connection_request::ConnectionRequest {
-    let mut address_info = NodeAddress::new();
-    address_info.host = host.to_string().into();
-    address_info.port = port;
-    let addresses_info = vec![address_info];
     let mut connection_request = connection_request::ConnectionRequest::new();
-    connection_request.addresses = addresses_info;
-    connection_request.tls_mode = if use_tls {
-        connection_request::TlsMode::SecureTls
-    } else {
-        connection_request::TlsMode::NoTls
+
+    connection_request.addresses = node_addresses_to_proto((*config).addresses, (*config).address_count as usize);
```

Comment:
We should probably take the config's `address_count` as a `usize` instead of doing this cast from a `u32`. Surely C# has a way to pass pointer sized integers across the FFI boundary (Golang can).